rfcs/modules/0001-pkg:list_pkgs-output.md

```diff
@@ -0,0 +1,297 @@
+
+### Proposed Design
+
+Proposed change is to follow _any_ Package Manager spirit: just return
+a list of packages. The new approach seeing the current re-mixing such
+list into a map only as a (custom/app)-level format and actually
+avoids this.
+
+Structural example would be as follows:
+
+```
+[
+	{
+		"name": "package-name",
+		"key": "value",
+	}
+]
+```
+
+That is, it would be just a list of dictionaries with the standard
+keys inside. Reworked output from the examples above would be like
+this:
+
+```
+[
+	{
+		"name": "xz-libs",
+		"release": "1.el7",
+		"install_date": "2017-06-24T10:51:34Z",
+		"version": "5.2.2",
+		"arch": "i686",
+		"install_date_time_t": 1498301494
+	},
+	{
+		"name": "xz-libs",
+		"release": "1.el7",
+		"install_date": "2017-06-24T10:51:34Z",
+		"version": "5.2.2",
+		"arch": "x86_64",
+		"install_date_time_t": 1498302356
+	}
+],
+
+```
+
+In case with the less attributes as follows:
+
+```
+[
+	{
+		"name": "kernel-default",
+		"release": "92.56.1",
+		"version": "4.4.103"
+	},
+	{
+		"name": "kernel-default",
+		"release": "69.1",
+		"version": "4.4.21"
+	}
+], 
+
+```
+
+This would by design invalidate all the issues with the naming,
+architecture and parsing all that accordingly. API consumer do not
+need anymore what exactly system it works with, as there would be a
+standard format across everywhere.
+
+The rest of the functinality would stay the same.
+
+### Notes
+
+Important to have a common validator decorator that would make sure
+_all the keys_ are always the same across all the operating systems.
+
+## Alternatives
+[alternatives]: #alternatives
+
+Alternatively it would be possible to just add parameter `delimeter`
+to the existing code and so the API consumer would not need to know
+which system it is talking to, in order to separate prepended part
+(architecture). For example, add a `/` (slash), so the return is like:
+
+```
+"xz-libs/i686": ....
+"xz-libs": ....
+```
+
+However, this does not solve two issues:
+
+1. There is no mention in the name what is the default architecture,
+   unless explicitly requested, since prepended architecture is only
+   different from the default. We only _assume_ that _most of the
+   time_ it is `x86_64` architecture which is capable running `i686`
+   or older.
+
+2. It is still needs to know there were no slashes introduced in the
+   naming of the packages to that particular operating system flavour.
+
+3. It will likely lead to a funny random delimeters.
+
+All this more like a workaround of the problem, instead of actally
+solving it.
+
+## Unresolved questions
+[unresolved]: #unresolved-questions
+
+N/A
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This change should not go to the prepared Oxygen release, but only in
+`develop` branch.
+
+- Implementation should be done using deprecation model system
+  (decorators we already have in Salt core).
+
+- The old version should stay for two releases (standard retention time)
```

Comment:
So there would be no changes out of the box until Magnesium, correct?

Reply:
I'd say, we need to maintain at least two versions backward-compatible in everything we change so drastically.

Reply:
I thkn that the way to do this would be to make the new function `pkg.list` and place a deprecation warning on list_pkgs.